### PR TITLE
[refactor](mem_reuse) refactor mem_reuse in MutableBlock

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/core/block.h [hunk 1]
```diff
@@ -413,34 +413,42 @@ class MutableBlock {
 
     using IndexByName = phmap::flat_hash_map<String, size_t>;
     IndexByName index_by_name;
+    Block* _output_block = nullptr;
```
Comment (HappenLee) [inline be/src/vec/core/block.h:416]: do not use this way to swap

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run arm

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.89 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          30.0 seconds inserted 10000000 Rows, about 333K ops/s
 storage size: 17159081262 Bytes

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.48 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17164245422 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.72 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          30.1 seconds inserted 10000000 Rows, about 332K ops/s
 storage size: 17162459694 Bytes

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.82 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162475271 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.41 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17165516996 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.58 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17170061547 Bytes

**Pull request merged**